src/transformState.js

```diff
@@ -5,7 +5,50 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const key in actions) {
+    const operationTypeObject = { ...actions[key] };
+
+    for (const operationKey in operationTypeObject) {
+      switch (operationTypeObject[operationKey]) {
+        case 'addProperties': {
+          addProperties(state, operationTypeObject.extraData);
```

Comment:
You can destructure data that you need  from operatioTypeObject.extraData, it will make your code look neat and clear